### PR TITLE
change list_stargazers to use the Github relevant API option with `starred_at`

### LINKED ISSUE #268: How to get the "starred_at" field with list_stargazers?
Github's ListGazer supports fetching the time when a gazer starred a repo (https://docs.github.com/en/rest/activity/starring#list-repositories-being-starred).

For that need to add an Accept Header to the request and then the response structure changes, so instead of a list of users:
`
[
  {
    "login": "octocat",
    "id": 1,
    ...
  }
  ...
]
`

it responds with a list that looks like this:
`
[
  {
    "starred_at": "2011-01-16T19:06:43Z",
    "user": {
      "login": "octocat",
      "id": 1,
      ...
   }
   ...
]
`

How can I access this different method of calling to retrieve the "starred_at" field?

I'm also good with extending Octocrab for this but couldn't figure out how to do it.

Thanks

### PR DESCRIPTION
**Pull request by yanshay**
Resolved #268 

list_stargazer now returns a new model StarGazer which includes starred_at
added get_with_headers and _get_with_headers to allow injecting headers (original get and _get call the new methods)
modified stargazers test to the change

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -698,25 +698,60 @@ impl Octocrab {
         P: Serialize + ?Sized,
         R: FromResponse,
     {
-        let response = self._get(self.absolute_url(route)?, parameters).await?;
-        R::from_response(crate::map_github_error(response).await?).await
+        // let response = self._get(self.absolute_url(route)?, parameters).await?;
+        // R::from_response(crate::map_github_error(response).await?).await
```
Comment (lineville) [inline src/lib.rs:702]: nit: dead code
```suggestion

```

File: src/lib.rs [hunk 2]
```diff
@@ -698,25 +698,60 @@ impl Octocrab {
...
         url: impl reqwest::IntoUrl,
         parameters: Option<&P>,
+    ) -> Result<reqwest::Response> {
+        // let mut request = self.client.get(url);
+
+        // if let Some(parameters) = parameters {
+        //     request = request.query(parameters);
+        // }
+
+        // self.execute(request).await
```
Comment (lineville) [inline src/lib.rs:718]: nit: dead code
```suggestion

```

File: tests/resources/stargazers.json [hunk 3]
```diff
@@ -1,29 +1,23 @@
...
-  "followers": 11,
-  "following": 14,
-  "created_at": "2017-01-25T16:04:54Z",
-  "updated_at": "2022-07-14T22:14:13Z"
-}
\ No newline at end of file
+  {
+      "starred_at": "2016-07-04T08:30:31Z",
+      "user": {
+          "login": "germtb",
```
Comment (lineville) [inline tests/resources/stargazers.json:4]: You may want to anonymize this a bit and just replace `germtb` with `user` or something less identifying

**Review by lineville — APPROVED** [review]
LGTM, thanks for the contribution ✨ -- can't merge it myself but looks good

File: tests/resources/stargazers.json (hunk 3 above)
Comment (yanshay) [reply tests/resources/stargazers.json]: Thanks, done.

File: src/api/repos/stargazers.rs [hunk 4]
```diff
@@ -32,12 +32,16 @@ impl<'octo, 'r> ListStarGazersBuilder<'octo, 'r> {
...
+  pub async fn send(self) -> crate::Result<crate::Page<crate::models::StarGazer>> {
     let url = format!(
       "repos/{owner}/{repo}/stargazers",
       owner = self.handler.owner,
       repo = self.handler.repo
     );
-    self.handler.crab.get(url, Some(&self)).await
+
+    let mut headers = reqwest::header::HeaderMap::new();
+    headers.insert(ACCEPT, "application/vnd.github.v3.star+json".parse().unwrap());
```
Comment (MarcoFalke) [inline src/api/repos/stargazers.rs:43]:
```suggestion
    headers.insert(ACCEPT, "application/vnd.github.star+json".parse().unwrap());
```

nit (feel free to ignore): While it does not affect behaviour, the docs recommend to not use `v3`. See https://docs.github.com/en/rest/activity/starring#custom-media-types-for-starring and https://docs.github.com/en/rest/overview/media-types ("Note: In the past, we recommended including v3 in your Accept header. This is no longer required and will have no impact on your API requests.")

**Comment by XAMPPRocky** [comment]
Thank you for your PR, and congrats on your first contribution! 🎉

**Pull request merged**